Add the WCCOM Account Icon to wc-admin home

bor0 opened the pull request:
### Changes proposed in this Pull Request:

<!-- If necessary, indicate if this PR is part of a bigger feature. Add a label with the format `focus: name of the feature [team:name of the team]`. -->

<!-- Describe the changes made to this Pull Request and the reason for such changes. -->

Closes https://github.com/Automattic/woocommerce.com/issues/23080.

<!-- Begin testing instructions -->

### How to test the changes in this Pull Request:

<!-- Include detailed instructions on how these changes can be tested. Review and follow the guide for how to write high-quality testing instructions. -->

Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

Run `localStorage.setItem( 'wccom-tracks-debug', true );` in the console and make sure you show verbose messages as well, to see track requests.

See [this recording](https://github.com/Automattic/woocommerce.com/issues/23080#issuecomment-2657421432) to get a vague idea.

1. Make sure you disconnect your WCCOM connection.
2. Go to `/wp-admin/admin.php?page=wc-admin`. Click on the account icon in the top-right, it should show a possibility to connect.
    - You should see a track request `wcadmin_header_account_click` (page wc-admin)
    - You should see the same possibility to connect in `/wp-admin/admin.php?page=wc-admin&path=%2Fextensions`, with an extra link to take you to wccom dashboard
        - You should see a track request `wcadmin_header_account_click` (page wc-addons)
4. Ensure you can make a connection to WCCOM from `/wp-admin/admin.php?page=wc-admin`.
    - You should also see a track request `wcadmin_header_account_connect_click` (page wc-admin)
    - Avatar should change to your image when you connect.
    - You should get redirected to `wc-admin` home (where you initiated the connection)
5. Ensure that you see your avatar and that you're connected when you also visit `/wp-admin/admin.php?page=wc-admin&path=%2Fextensions`
6. Go back to `/wp-admin/admin.php?page=wc-admin`, and try to disconnect.
    - You should get redirected to `wc-admin` home (where you initiated the disconnection)
7. Ensure that you are disconnected when you also visit `/wp-admin/admin.php?page=wc-admin&path=%2Fextensions`
8. Make sure you can also connect/disconnect from `/wp-admin/admin.php?page=wc-admin&path=%2Fextensions`, and that you are redirected back properly

<!-- End testing instructions -->

### Changelog entry

<!-- You can optionally choose to enter a changelog entry by checking the box below and supplying data. -->
<!-- It will trigger the 'Add changelog to PR' CI job to create and push the entry into the branch. -->

<!-- Due to org permissions, the job might fail for PRs crated from a fork under GitHub organizations. Possible solutions: -->
<!-- * Create entry manually with `pnpm --filter='@woocommerce/plugin-woocommerce' changelog add` and push it into the branch (replace `@woocommerce/plugin-woocommerce` with package name from nearest `package.json` file) -->
<!-- * Create entry from supplied PR data and push it automatically `pnpm utils changefile pr-number-here -o github-org-name-here` -->

-   [ ] Automatically create a changelog entry from the details below.

<!-- If no changelog entry is required for this PR, you can specify that below and provide a comment explaining why. This cannot be used if you selected the option to automatically create a changelog entry above. -->

-   [ ] This Pull Request does not require a changelog entry. (Comment required below)

<details>

<summary>Changelog Entry Details</summary>

#### Significance

<!-- Choose only one -->

-   [ ] Patch
-   [ ] Minor
-   [ ] Major

#### Type

<!-- Choose only one -->

-   [ ] Fix - Fixes an existing bug
-   [ ] Add - Adds functionality
-   [ ] Update - Update existing functionality
-   [ ] Dev - Development related task
-   [ ] Tweak - A minor adjustment to the codebase
-   [ ] Performance - Address performance issues
-   [ ] Enhancement - Improvement to existing functionality

#### Message <!-- Add a changelog message here -->

</details>

<details>

<summary>Changelog Entry Comment</summary>

#### Comment <!-- If your Pull Request doesn't require a changelog entry, a comment explaining why is required instead -->

</details>

bor0 committed:
Add the WCCOM Account Icon to wc-admin home

bor0 commented on plugins/woocommerce/includes/admin/helper/class-wc-helper.php:
```diff
@@ -51,6 +51,16 @@ public static function load() {
 		do_action( 'woocommerce_helper_loaded' );
```
All of the changes in this file is to support different redirect_uri - either from wc-addons (Extensions) or wc-admin (Home)

bor0 commented on plugins/woocommerce/includes/admin/helper/class-wc-helper-admin.php:
```diff
@@ -28,12 +28,11 @@ class WC_Helper_Admin {
-			$is_in_app_marketplace = (
+			$is_wc_home_or_in_app_marketplace = (
 				isset( $_GET['page'] ) && 'wc-admin' === $_GET['page'] //phpcs:ignore WordPress.Security.NonceVerification.Recommended
-				&& isset( $_GET['path'] ) && '/extensions' === $_GET['path'] //phpcs:ignore WordPress.Security.NonceVerification.Recommended
```
Account for both `wc-admin` (home) and `wc-addons` (Extensions)

bor0 commented on plugins/woocommerce/client/admin/client/marketplace/components/header-account/header-account.tsx:
```diff
@@ -27,13 +28,19 @@ interface MenuItemProps extends ComponentProps< typeof OriginalMenuItem > {
+}
+
+export default function HeaderAccount( {
+	page = 'wc-admin',
```
I don't like this part too much. The component changes behavior based on the page it comes from, but at the same time, I didn't feel like creating a duplicate component here.

Not sure if there's a cleaner way.

bor0 committed:
Lint

KokkieH committed:
Merge branch 'trunk' into add/account-icon-wc-admin-home

KokkieH commented on plugins/woocommerce/client/admin/client/marketplace/components/header-account/header-account.tsx:
```diff
@@ -109,20 +124,28 @@ export default function HeaderAccount(): JSX.Element {
 								href={ accountOrConnect }
+								onClick={ () =>
+									recordEvent(
+										'header_account_connect_click',
```
So this is the one that's firing when clicking on the email address when already connected.

KokkieH commented on plugins/woocommerce/client/admin/client/marketplace/components/header-account/header-account.tsx:
```diff
@@ -109,20 +124,28 @@ export default function HeaderAccount(): JSX.Element {
+										className="woocommerce-marketplace__menu-icon"
+									/>
+									{ __(
+										'WooCommerce.com account',
+										'woocommerce'
```
I don't think it's necessary to show this once we're already connected, as it duplicates the link attached to the email address.

KokkieH reviewed (commented):
Works beautifully!

Just some thoughts re Tracks events:

* Once connected, if I click on my email address it again fires `wcadmin_header_account_connect_click`, the same event that fires when clicking to start the connection flow when disconnected. Should we perhaps add a new event here, e.g. `wcadmin_header_account_view_click`?
* Can we add a new event for clicking on the Disconnect option? It doesn't appear to have one at present, from either `wc-admin` or `wc-addons`.
* The dashboard link on `wc-addons` also has no tracks event. The same `wcadmin_header_account_view_click` I suggested for clicking the email would work here as well, as the two go to the same place. Though perhaps we should hide it if the site is connected in favour of the email link?

The first one I think we should fix before shipping this, else it will inflate instances of the `wcadmin_header_account_connect_click` by recording clicks from both new connections and already-connected users.

The others, while I think it's worthwhile to record that data, are non-blocking.

bor0 committed:
Some tweaks around tracking events

- Distinction between `header_account_view_click` (connected) and `header_account_connect_click` (disconnected)
- Hide duplicate my-dashboard link when connected, and also add `header_account_view_click` to that link
- Another event introduced for disconnection `header_account_disconnect_click`

Props @KokkieH

KokkieH reviewed (approved):
Thanks for the changes. Confirmed the new events fire as expected, and there's no longer a duplicate dashboard link on IAM when connected, but it's still showing when disconnected.

bor0 commented:
Thanks for the review! Good catch with the tracks
